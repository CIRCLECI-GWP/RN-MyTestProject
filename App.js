import React from 'react';
import {
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  Button,
  TextInput,
  View,
} from 'react-native';

import {Colors} from 'react-native/Libraries/NewAppScreen';

const App = () => {
  const [message, setMessage] = React.useState();

  const isDarkMode = useColorScheme() === 'dark';
  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor={backgroundStyle.backgroundColor}
      />

      <Button
        title="Say Hello"
        onPress={() => {
          setTimeout(() => {
            setMessage('Hello Tester');
          }, Math.floor(Math.random() * 200));
        }}
      />
      {message && (
        <Text style={styles.messageText} testID="printed-message">
          {message}
        </Text>
      )}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  messageText: {
    fontFamily: 'Arial',
    fontSize: 38,
    textAlign: 'center',
    marginTop: 10,
  },
});

export default App;
