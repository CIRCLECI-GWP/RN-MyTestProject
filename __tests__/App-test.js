import 'react-native';
import React from 'react';
import App from '../App';

import {fireEvent, render, waitFor} from '@testing-library/react-native';

it('Renders Message', async () => {
  const {getByTestId, getByText, queryByTestId, toJSON} = render(<App />);

  const button = getByText('Say Hello');
  fireEvent.press(button);

  await waitFor(() => expect(queryByTestId('printed-message')).toBeTruthy());

  expect(getByTestId('printed-message').props.children).toBe('Hello Tester');
  expect(toJSON()).toMatchSnapshot();
});
