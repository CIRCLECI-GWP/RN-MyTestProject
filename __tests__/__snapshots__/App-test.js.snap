// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Message 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Say Hello
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "fontFamily": "Arial",
        "fontSize": 38,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
    testID="printed-message"
  >
    Hello Tester
  </Text>
</RCTSafeAreaView>
`;
